services/web/server/src/simcore_service_webserver/catalog_client.py

```diff
@@ -0,0 +1,124 @@
+""" Requests to catalog service API
+
+"""
+import logging
+import urllib.parse
+from typing import Any, Dict, List, Optional
+
+from aiohttp import web
+from aiohttp.client_exceptions import ClientConnectionError, ClientResponseError
+from servicelib.rest_responses import wrap_as_envelope
+from yarl import URL
+
+from ._meta import api_version_prefix
+from .catalog_config import KCATALOG_ORIGIN, KCATALOG_VERSION_PREFIX, get_client_session
+from .constants import X_PRODUCT_NAME_HEADER
+
+logger = logging.getLogger(__name__)
+
+
+async def is_service_responsive(app: web.Application):
+    """ Returns true if catalog is ready """
+    origin: Optional[URL] = app.get(KCATALOG_ORIGIN)
+
+    if not origin:  # service was not enabled!
+        return False
+
+    client = get_client_session(app)
+
+    # call to health-check entry-point
+    async with client.get(origin, ssl=False) as resp:
+        return resp.status == 200
+
+
+def to_backend_service(rel_url: URL, origin: URL, version_prefix: str) -> URL:
+    """Translates relative url to backend catalog service url
+
+    E.g. https://osparc.io/v0/catalog/dags -> http://catalog:8080/v0/dags
+    """
+    assert not rel_url.is_absolute()  # nosec
+    new_path = rel_url.path.replace(
+        f"/{api_version_prefix}/catalog", f"/{version_prefix}"
+    )
+    return origin.with_path(new_path).with_query(rel_url.query)
+
+
+async def make_request_and_envelope_response(
+    app: web.Application,
+    method: str,
+    url: URL,
+    headers: Optional[Dict[str, str]] = None,
+    data: Optional[bytes] = None,
+) -> web.Response:
+    """
+    Helper to forward a request to the catalog service
+    """
+    session = get_client_session(app)
+
+    try:
+
+        async with session.request(method, url, headers=headers, data=data) as resp:
+            payload = await resp.json()
+
+            try:
+                resp.raise_for_status()
+                resp_data = wrap_as_envelope(data=payload)
+
+            except ClientResponseError as err:
+                if 500 <= err.status:
+                    raise err
+                resp_data = wrap_as_envelope(error=payload["errors"])
+
+            return web.json_response(resp_data, status=resp.status)
+
+    except (TimeoutError, ClientConnectionError, ClientResponseError) as err:
+        logger.warning(
+            "Catalog service errors upon request %s %s: %s", method, url.relative(), err
+        )
+        raise web.HTTPServiceUnavailable(
+            reason="catalog is currently unavailable"
+        ) from err
```

Comment:
I think here we might have the same issue with exception handling as for the states no? if one of those calls fail? what happens when listing projects?

Reply:
this should be consider together with the other clients. I created #2156.